subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/FlakinessIssueReporter.groovy

```diff
@@ -0,0 +1,110 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.results
+
+import groovy.transform.CompileStatic
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.gradle.ci.common.model.FlakyTest
+import org.gradle.ci.github.GitHubIssuesClient
+import org.gradle.ci.tagging.flaky.KnownFlakyTestProvider
+import org.gradle.performance.results.ScenarioBuildResultData.ExecutionData
+import org.kohsuke.github.GHIssue
+import org.kohsuke.github.GHIssueState
+
+import static org.gradle.ci.github.GitHubIssuesClient.CI_TRACKED_FLAKINESS_LABEL
+import static org.gradle.ci.github.GitHubIssuesClient.FROM_BOT_PREFIX
+import static org.gradle.ci.github.GitHubIssuesClient.MESSAGE_PREFIX
+import static org.gradle.ci.github.GitHubIssuesClient.TEST_NAME_PREFIX
+
+@CompileStatic
+class FlakinessIssueReporter {
+    static final String GITHUB_FIX_IT_LABEL = "fix-it"
+    private final KnownFlakyTestProvider provider
+    private final GitHubIssuesClient gitHubIssuesClient
+
+    FlakinessIssueReporter(GitHubIssuesClient gitHubIssuesClient, KnownFlakyTestProvider provider) {
+        this.provider = provider
+        this.gitHubIssuesClient = gitHubIssuesClient
+    }
+
+    void report(ScenarioBuildResultData flakyScenario) {
+        FlakyTest knownFlakyTest = findKnownFlakyTest(flakyScenario)
+        if (knownFlakyTest) {
+            if (issueClosed(knownFlakyTest)) {
+                knownFlakyTest.issue.reopen()
+            }
+            if (!hasFixItLabel(knownFlakyTest)) {
+                knownFlakyTest.issue.addLabels(GITHUB_FIX_IT_LABEL)
+            }
+        } else {
+            knownFlakyTest = openNewFlakyTestIssue(flakyScenario)
+        }
+
+        commentCurrentFailureToIssue(flakyScenario, knownFlakyTest.issue)
+    }
+
+    @TypeChecked(TypeCheckingMode.SKIP)
+    static void commentCurrentFailureToIssue(ScenarioBuildResultData scenario, GHIssue issue) {
+        issue.comment("""
+${FROM_BOT_PREFIX}
+
+Coordinator url: https://builds.gradle.org/viewLog.html?buildId=${System.getenv("BUILD_ID")}
+Worker url: ${scenario.webUrl}
+Agent: [${scenario.agentName}](${scenario.agentUrl})
+Details:
+
+| Iteration | Difference | Confidence |
+|---|---|---|
+${assembleTable(scenario)}
+""")
+    }
+
+    private static String assembleTable(ScenarioBuildResultData scenario) {
+        scenario.executions.withIndex().collect { ExecutionData execution, int index ->
+            "|${index + 1}|${execution.differenceDisplay}|${execution.formattedConfidence}|"
+        }.join('\n')
+    }
+
+    private FlakyTest openNewFlakyTestIssue(ScenarioBuildResultData flakyScenario) {
+        String testName = flakyScenario.scenarioName
+        String title = "Flaky performance test: ${testName}"
+        String message = "Flaky performance test scenario"
```

Comment:
Why should we check for this message? Shouldn't the message be something like `Performance test failed` or whatever is printed in the log so we tag the flaky tests which happen during the week as well?

Reply:
We don't check this message. This is just a string to avoid https://github.com/gradle/gradle-private/issues/1967 .

Reply:
Hmm... Did we stop checking the message for flaky tests at some point?

Reply:
No, sorry if I didn't make myself clear, I just wasn't reusing this code: https://github.com/gradle/ci-health/blob/b520838254bb6845e6847fc578fbd1c1651bac2c/tagging/src/main/groovy/org/gradle/ci/tagging/flaky/InvalidFailureErrorAnalyzer.groovy#L24

Reply:
I think we should re-use that code. I don't see why we want to handle flaky performance test failures much different from other flaky test failures.

Reply:
Now I'm using a message `we're slower than` so the future failure message can match it.